src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -23,6 +23,180 @@ namespace System
+
+                case StringComparison.Ordinal:
+                    if (span.Length == 0 || value.Length == 0)
+                        return span.Length - value.Length;
+                    return CompareOrdinalHelper(span, value);
+
+                case StringComparison.OrdinalIgnoreCase:
+                    return CompareInfo.CompareOrdinalIgnoreCase(span, value);
+
+                default:
+                    throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+            }
+        }
+
+        public static unsafe int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparisonType, out int matchedLength)
+        {
+            if (value.Length == 0)
+            {
+                StringSpanHelpers.CheckStringComparison(comparisonType);
+                matchedLength = 0;
+                return 0;
+            }
+
+            if (span.Length == 0)
+            {
+                StringSpanHelpers.CheckStringComparison(comparisonType);
+                matchedLength = 0;
+                return -1;
+            }
+
+            int defaultMatchLength = value.Length;
+
+            switch (comparisonType)
+            {
+                case StringComparison.CurrentCulture:
+                    int index = IndexOfCultureHelper(span, value, &defaultMatchLength);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.CurrentCultureIgnoreCase:
+                    index = IndexOfCultureIgnoreCaseHelper(span, value, &defaultMatchLength);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.InvariantCulture:
+                    index = IndexOfCultureHelper(span, value, &defaultMatchLength, invariantCulture: true);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.InvariantCultureIgnoreCase:
+                    index = IndexOfCultureIgnoreCaseHelper(span, value, &defaultMatchLength, invariantCulture: true);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.Ordinal:
+                    matchedLength = defaultMatchLength;
+                    return IndexOfOrdinalHelper(span, value, false);
+
+                case StringComparison.OrdinalIgnoreCase:
+                    matchedLength = defaultMatchLength;
+                    return IndexOfOrdinalHelper(span, value, true);
+
+                default:
+                    throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+            }
+        }
+
+        internal static unsafe int IndexOfCultureHelper(ReadOnlySpan<char> span, ReadOnlySpan<char> value, int* matchLengthPtr, bool invariantCulture = false)
```

Comment:
>bool invariantCulture = false [](start = 128, length = 29)

you don't need this. you already checking this value, instead you can just check CompareInfo.Name.Length == 0. remember current culture can be invariant too :-)

Reply:
> you don't need this. 

What are you referring to?

I don't understand what you mean.

Reply:
you don't need the parameter invariantCulture. 

just check CompareInfo.Name.Length == 0 inside the method

---
In reply to: [169813408](https://github.com/dotnet/coreclr/pull/16467#discussion_r169813408) [](ancestors = 169813408)